example/thirdparty/ollama-js/build.mill

```diff
@@ -0,0 +1,224 @@
+
+    os.write.over(
+      Task.dest / "package.json",
+      ujson.Obj(
+        "name" -> "ollama",
+        "version" -> "0.0.0",
+        "description" -> "Ollama Javascript library",
+        "main" -> "dist/index.cjs",
+        "module" -> "dist/index.mjs",
+        "types" -> "dist/index.d.ts",
+        "exports" -> ujson.Obj(
+          "." -> ujson.Obj(
+            "require" -> "./dist/index.cjs",
+            "import" -> "./dist/index.mjs",
+            "types" -> "./dist/index.d.ts"
+          ),
+          "./browser" -> ujson.Obj(
+            "require" -> "./dist/browser.cjs",
+            "import" -> "./dist/browser.mjs",
+            "types" -> "./dist/browser.d.ts"
+          ),
+          "./*" -> "./*"
+        ),
+        "dependencies" -> ujson.Obj("whatwg-fetch" -> "3.6.20")
+      )
+    )
+
+    os.call("node_modules/.bin/unbuild", cwd = T.dest)
+
+    // prepare bundled ouptut to be used as `unmanagedDeps` in examples section.
+    os.makeDir.all(T.dest / "ollama")
+    os.copy(T.dest / "package.json", T.dest / "ollama" / "package.json")
+    os.copy(T.dest / "dist", T.dest / "ollama" / "dist")
+
+    PathRef(T.dest / "ollama")
+  }
+}
+
+object `package` extends RootModule with OllamaModule with TsLintModule {
+  def moduleName = "core"
+  def npmDeps = Seq("whatwg-fetch@3.6.20")
+  def npmDevDeps = Seq(
+    "@swc/core@1.10.12",
+    "@types/whatwg-fetch@0.0.33",
+    "unbuild@2.0.0"
+  )
+
+  def npmLintDeps = Seq(
+    "prettier@3.2.4",
+    "eslint@8.29.0",
+    "@typescript-eslint/eslint-plugin@5.42.1",
+    "@typescript-eslint/parser@5.42.1"
+  )
+
+  // disable declarations output in compile stage
+  def runTypeCheck: T[Boolean] = false
+
+  def options =
+    super.options() + (
+      "ts-node" -> ujson.Obj(
+        "swc" -> ujson.True,
+        "esm" -> ujson.True
+      )
+    )
+
+  def compilerOptions =
+    super.compilerOptions() + (
+      "noImplicitAny" -> ujson.True,
+      "noImplicitThis" -> ujson.True,
+      "strictNullChecks" -> ujson.True,
+      "declarationMap" -> ujson.True,
+      "skipLibCheck" -> ujson.True,
+      "strict" -> ujson.True,
+      "forceConsistentCasingInFileNames" -> ujson.True,
+      "module" -> ujson.Str("ES2022"),
+      "target" -> ujson.Str("es6"),
+      "lib" -> ujson.Arr.from(Seq("es6", "es2018.asyncgenerator", "dom"))
+    )
+
+  object test extends TypeScriptTests with TestModule.Vitest
+
+  object examples extends TypeScriptModule {
+    object abort extends TypeScriptModule {
+      def unmanagedDeps = Seq(unbuild())
+      def sources = Task.Sources(moduleDir)
+      def mainFileName = s"abort-all-requests.ts"
+      def mainFilePath = Task { compile().path / mainFileName() }
+    }
+
+    object `fill-in-middle` extends TypeScriptModule {
+      def unmanagedDeps = Seq(unbuild())
+      def sources = Task.Sources(moduleDir)
+      def mainFileName = s"fill.ts"
+      def mainFilePath = Task { compile().path / mainFileName() }
+      def enableEsm = true
```

Comment:
The boilerplate that is common all the examples should be moved into an `trait ExampleModule extends TypeScriptModule` that all the examples can inherit from